File: Session15/198. House Robber.java
class Solution {
    public int rob(int[] nums) {
        int n=nums.length;
        
        int dp[]=new int[n+2];
        dp[0]=0;
        dp[1]=0;
        
        for(int i=2;i<n+2;i++)
        {
            dp[i]=Math.max(nums[i-2]+dp[i-2], dp[i-1]);
        }
        
        return dp[n+1];
    }
}

File: Session15/198. House Robber.py
class Solution:
    def rob(self, A):
        if len(A) == 1: return A[0]
        dp = (len(A)+2)*[0]
        dp[0]=0
        dp[1] =0
        for i in range(2, len(A)+2):
            dp[i] = max(dp[i-1], A[i-2] + dp[i-2])
        return dp[-1]